.github/workflows/diff.yaml

```diff
@@ -75,14 +75,30 @@ jobs:
       - name: Fetch all history for all tags and branches
         run: git fetch
 
-      - name: Build combined ASAN, UBSAN and coverage binaries
+      - name: Initialize deps
         run: |
           # Activate toolchain.
           source /opt/toolchain-v4/activate
 
           # Initialize dependencies.
           ./init
 
+      - name: Python code analysis
+        run: |
+          CHANGED_FILES=$(git diff -U0 HEAD~1... --name-only)
```

Comment:
Wouldn't this only contain the diffs from the latest commit? I think we should use something similar to how we run `clang-tidy`: `git diff -U0 ${{ env.BASE_BRANCH }}...`. Check below how it is used.